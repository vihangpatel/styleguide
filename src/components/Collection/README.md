With Loading

```js
<Device>
	<Collection
		title="Discover"
		description="Events Happening Now"
		cards={[]}
		loading
	/>
</Device>
```
With actual data

```js
<Device>
	<Collection
		title="Discover"
		description="Events Happening Now"
		cards={
			[
				{
					name: "Today",
					description: "Events happening today",
					icon: "today",
					image:"http://in.bmscdn.com/expcards/weekend_planner_24-jul-2015_07-47.jpg",
					href:"http://in.bmscdn.com/expcards/weekend_planner_24-jul-2015_07-47.jpg"
				},
				{
					name: "Tomorrow",
					description: "Events happening tomorrow",
					icon: "tomorrow",
					image:"http://in.bmscdn.com/expcards/weekend_planner_24-jul-2015_07-47.jpg"
				},
				{
					name: "Weekend",
					description: "Events happening over the weekend",
					icon: "weekend",
					image: "http://in.bmscdn.com/expcards/weekend_planner_24-jul-2015_07-47.jpg"
				}
			]
		}
	/>
</Device>
```