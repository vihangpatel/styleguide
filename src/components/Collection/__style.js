import { margin, padding } from 'polished'

export const __wrapper = () => ({
	backgroundColor: '#FFFFFF',
	...padding(8, 0),
	...margin(0, 0, 16, 0),
	width: '100%',
})

export const __header = props => ({
	fontSize: props.theme.fontSize.xl,
	marginBottom: 5,
	marginLeft: 16,
	...props.theme.mediumBaseFont,
})

export const __description = props => ({
	fontSize: props.theme.fontSize.s,
	color: props.theme.color.darkGrey,
	marginBottom: 13,
	marginLeft: 16,
	...props.theme.regularFont,
})

export const __collectionScroller = () => ({
	display: 'flex',
	alignItems: 'flex-start',
	overflowX: 'scroll',
	marginBottom: 16,
	'&::-webkit-scrollbar': {
		display: 'none',
	},
})

export const __innerScroll = () => ({
	display: 'flex',
	alignItems: 'flex-start',
	maxWidth: 'fit-content',
})

export const __cardKeyframe = () => ({
	'0%': { opacity: '0.3' },
	'50%': { opacity: '1' },
	'100%': { opacity: '0.3' },
})

export const __placeHolder = props => ({
	width: `${props.width || 'auto'}`,
	height: `${props.height || '240'}`,
	background: '#f2f2f2',
	animationName: __cardKeyframe(),
	animationDuration: '1s',
	animationTimingFunction: 'linear',
	animationIterationCount: 'infinite',
	boxSizing: 'border-box',
})

export const __card = props => ({
	width: 156,
	height: 156,
	marginRight: 16,
	backgroundColor: '#f2f2f2',
	borderRadius: props.theme.borderRadius,
	extend: {
		condition: props.loading === true,
		style: {
			animationName: __cardKeyframe(),
			animationDuration: '1s',
			animationTimingFunction: 'linear',
			animationIterationCount: 'infinite',
		},
	},
	'&:first-child': {
		marginLeft: 16,
	},
})
