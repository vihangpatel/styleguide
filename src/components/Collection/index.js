import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { createComponent } from 'react-fela'

import CollectionStack from './components/CollectionStack'
import LoadingCollection from './components/LoadingCollection'

import { __header, __description, __wrapper } from './__style'

const Header = createComponent(__header)
const Description = createComponent(__description)
const Wrapper = createComponent(__wrapper)

class Collection extends PureComponent {
	static propTypes = {
		/** Pass Array of colection data */
		cards: PropTypes.arrayOf(PropTypes.object),
		/** Title of Collection */
		title: PropTypes.string,
		/** Description of collection */
		description: PropTypes.string,
		/** Show loading cards */
		loading: PropTypes.bool,
	}

	static defaultProps = {
		cards: [],
		title: '',
		description: '',
		loading: false,
	}

	render() {
		const { title, description, loading, cards } = this.props

		if (!loading && cards.length === 0) {
			return null
		}

		return loading ? (
			<LoadingCollection />
		) : (
			<Wrapper>
				{title &&
					description && (
						<div>
							<Header>{title}</Header>
							<Description>{description}</Description>
						</div>
					)}
				<CollectionStack cards={cards} />
			</Wrapper>
		)
	}
}

export default Collection
