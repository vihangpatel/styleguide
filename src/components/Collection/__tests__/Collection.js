import React from 'react'
import Collection from 'components/Collection'
import renderer from 'react-test-renderer'
import StyleProvider from 'utils/Provider'

describe('Collection Testing', () => {
	it('shallow renders correctly when data provided', () => {
		const collection = renderer
			.create(
				<StyleProvider>
					<Collection
						title="Discover"
						description="Events Happening Now"
						cards={[
							{
								name: 'Today',
								description: 'Events happening today',
								icon: 'today',
								imageBg:
									'//in.bmscdn.com/webin/test-m6/aggregation/today.jpg',
							},
							{
								name: 'Tomorrow',
								description: 'Events happening tomorrow',
								icon: 'tomorrow',
								imageBg:
									'//in.bmscdn.com/webin/test-m6/aggregation/tomorrow.jpg',
							},
							{
								name: 'Weekend',
								description:
									'Events happening over the weekend',
								icon: 'weekend',
								imageBg:
									'//in.bmscdn.com/webin/test-m6/aggregation/weekend.jpg',
							},
						]}
					/>
				</StyleProvider>
			)
			.toJSON()
		expect(collection).toMatchSnapshot()
	})

	it('shallow renders correctly when data is loading', () => {
		const collectionLoading = renderer
			.create(
				<StyleProvider>
					<Collection loading />
				</StyleProvider>
			)
			.toJSON()
		expect(collectionLoading).toMatchSnapshot()
	})
})
