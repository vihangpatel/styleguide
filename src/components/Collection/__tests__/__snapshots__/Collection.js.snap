// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Testing shallow renders correctly when data is loading 1`] = `
<div>
  <div
    className="a"
  >
    <div
      className="ak al am an ah ao ap aq"
    />
    <div
      className="ar as"
    />
    <div
      className="at au am an ah ao ap aq"
    />
  </div>
  <div
    className="av aw"
  />
  <div
    className="i j k l"
  >
    <div
      className="i j m"
    >
      <div
        className="n o p q r s an ah ao ap"
      />
      <div
        className="n o p q r s an ah ao ap"
      />
      <div
        className="n o p q r s an ah ao ap"
      />
    </div>
  </div>
</div>
`;

exports[`Collection Testing shallow renders correctly when data provided 1`] = `
<div>
  <div
    className="a"
  >
    <div>
      <div
        className="b c d e"
      >
        Discover
      </div>
      <div
        className="f g h"
      >
        Events Happening Now
      </div>
    </div>
  </div>
  <div
    className="i j k l"
  >
    <div
      className="i j m"
    >
      <div
        className="n o p q r s"
      >
        <div>
          <img
            alt="Events happening today"
            className="t ab v w x ac ae af ag ah ai aj"
            onClick={undefined}
            src="//in.bmscdn.com/webin/test-m6/aggregation/today.jpg"
          />
        </div>
      </div>
      <div
        className="n o p q r s"
      >
        <div>
          <img
            alt="Events happening tomorrow"
            className="t ab v w x ac ae af ag ah ai aj"
            onClick={undefined}
            src="//in.bmscdn.com/webin/test-m6/aggregation/tomorrow.jpg"
          />
        </div>
      </div>
      <div
        className="n o p q r s"
      >
        <div>
          <img
            alt="Events happening over the weekend"
            className="t ab v w x ac ae af ag ah ai aj"
            onClick={undefined}
            src="//in.bmscdn.com/webin/test-m6/aggregation/weekend.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
